Make Fundraisers CampaignMembers to their fundraising page Campaigns

Alexalex1432 opened the pull request:
Improving Campaign mapping for fundraisers, now the fundraiser will be added as a CampaignMember to their fundraising pages

Also adding Campaign & DonorEmail classes to package.xml, they were previously missing and when I pulled from the server got some unrelated or minor changes to those classes that weren't yet includes in git
Resolving FUN-6738

Beta link for the managed package including these changes:
https://login.salesforce.com/packaging/installPackage.apexp?p0=04t1C000000OD0S

Commit:
Improving Campaign mapping for fundraisers, now the fundraiser will be added as a CampaignMember to their fundraising pages

Also adding Campaign & DonorEmail classes to package.xml, they were previously missing and when I pulled from the server got some unrelated or minor changes to those classes that weren't yet includes in git
Resolving FUN-6738

funraise-brian commented on src/classes/frCampaign.cls:
```diff
@@ -76,27 +76,54 @@ public class frCampaign {
+        }
+        
+        String supporterId = String.valueOf(request.get('supporterId'));
+        if(String.isNotBlank(supporterId)) {
```
Should this use the same type of check as line 91?

```suggestion
        if(supporterId != null && supporterId != '0') {
```

In platform our SFDCCampaignMessage DTO has `parentGoalId` and `supporterId` both as primitive types: `long`

Alexalex1432 replied:
https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_methods_system_string.htm

We're good on checking if it's null since isBlank/isNotBlank will check:

> Returns true if the specified String is white space, empty (''), or null; otherwise, returns false.

don't think we can use this check on line 91 though because there's some logic there around it not being the string "0" which I'd like to reinvestigate why that's happening but in a different PR

Alexalex1432 replied:
Oh I checked the platform logic and you're right, in ToSFDCCampaignMessage we'll send 0 for the supporterId here if it isn't a fundraising page. I'll add that check then

funraise-brian commented on src/classes/frCampaign.cls:
```diff
@@ -76,27 +76,54 @@ public class frCampaign {
+            if(donor != null) {
+				Boolean campaignMemberExists = [SELECT COUNT() FROM CampaignMember 
+                                                WHERE ContactId = :donor.getContactId() 
+                                                AND CampaignId = :newCampaign.Id] > 0;
```
Looks like the spacing here is weird because there might be some mixed tabs and spacing

Alexalex1432 replied:
pushed a commit to fix this

Commit:
Fixing whitespace in frCampaign, using whatever SFDC developer console prefers.

Resolving FUN-6738

Commit:
Correct the check for supporterId since we may send "0" if there isn't a supporter associated with a fundraising page

change both checks to use isNotBlank instead of just null checks

Resolving FUN-6738

Alexalex1432 commented:
install link for package incorporating feedback here:
https://login.salesforce.com/packaging/installPackage.apexp?p0=04t1C000000OD16
